### DISCUSSION TITLE
Rendering text with escape sequences in tables

### DISCUSSION THREAD

**Discussion by PlasmaHH** [Textualize/rich#3567] [Q&A]
Hi, is it possible to somehow tell rich tables how much space some text is going to occupy when rendered when its different than the number of characters? I get some text with ansi escape sequences from other places and would like to output them but rich takes the length of the strings and makes the columns way too wide, along with messing up borders because it doesn't know that some of the characters don't occupy any space when rendered...

**Comment by TomJGooding** [comment]
>I get some text with ansi escape sequences from other places...

It sounds like you might want [`Text.from_ansi`](https://rich.readthedocs.io/en/stable/text.html)?

**Reply by PlasmaHH** [reply]
Thanks, that works but seems to slow down certain text heavy parts of my application (from 1.4s to 3.3s) ... upon skimming through the code it seems that this parses the escape sequences and takes what is left over... I already have the "display length", is there maybe a way I can just tell it?
